packages/neon-core/src/tx/transaction/Transaction.ts

```diff
@@ -289,8 +286,15 @@ export class Transaction {
   }
 
   public getScriptHashesForVerifying(): string[] {
-    let hashes = this.cosigners.map(cosigner => cosigner.account);
-    hashes.unshift(this.sender);
+    const hashes = this.cosigners.map(cosigner => cosigner.account);
+    if (
+      !hashes.every(hash => hashes.indexOf(hash) === hashes.lastIndexOf(hash))
+    ) {
+      throw new Error(`Duplicate Cosigner`);
```

Comment:
this looks like something we should be blocking on construction of transaction if this state is illegal.

Reply:
Done